apiserver/facades/agent/machine/machiner.go

```diff
@@ -139,3 +139,53 @@ func (api *MachinerAPI) Jobs(args params.Entities) (params.JobsResults, error) {
 	}
 	return result, nil
 }
+
+// RecordAgentStartTime updates the agent start time field in the machine doc.
+func (api *MachinerAPI) RecordAgentStartTime(args params.Entities) (params.ErrorResults, error) {
+	results := params.ErrorResults{
+		Results: make([]params.ErrorResult, len(args.Entities)),
+	}
+	canModify, err := api.getCanModify()
+	if err != nil {
+		return results, err
+	}
+	for i, entity := range args.Entities {
+		tag, err := names.ParseMachineTag(entity.Tag)
```

Comment:
Now we've added the third usage, can we break this trilogy of:
- parsing the tag
- checking `canModify`
- retrieving the machine
into a new method?

Reply:
Is this what you had in mind? https://github.com/juju/juju/pull/10902/commits/b9ba7cc3f7c94e7b6a28c808d8ae7520d59bb609